### PR TITLE
Broadcast Radio + Microphone

### PR DESCRIPTION
**Pull request by noctyrnal**
## About the PR

Adds a radio, which broadcasts messages sent from the Broadcast Microphone, akin to an actual radio, letting stations have their own radio shows and hosts. Ideally should be mapped in for reporters.

## Why / Balance

Gives more variety to reporters, and adds more decorative / "lively" aspects to places e.g a radio in the bar.

## Technical details

N/A

## Media

- [ X ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image

https://github.com/space-wizards/space-station-14/assets/99278536/69caf560-6889-4db7-b9ce-61b04471ff0c

## Breaking changes

N/A

**Changelog**

🆑
- add: Added the Radio and Broadcast microphone. Host your own radio shows today!

### CODE REVIEW

**Review by lzk228 — CHANGES_REQUESTED** [review]
they should be obtainable

**Comment by Hmeister-fake** [comment]
> they should be obtainable

arent these made for an upcoming role though? don't think they should be obtainable till the role for them is done

**Comment by noctyrnal** [comment]
> > they should be obtainable
> 
> arent these made for an upcoming role though? don't think they should be obtainable till the role for them is done

They've been made for reporters, though also as RP tools for any radio hosts / self-proclaimed radio hosts in the future. Made them craftable too, so people can decorate their spaces with it easier, and means mappers dont have to map in 500 for full radio host gaming.

You can now craft them with 3 Steel and 2 LV cables.

**Comment by Dutch-VanDerLinde** [comment]
changelog broken you need 🆑

**Comment by noctyrnal** [comment]
> changelog broken you need 🆑

Fixed on both PRs

**Comment by FairlySadPanda** [comment]
Chatfactor warning, I'll need to eyeball this code (similar to how I had to fuck with handheld radios) during the content refactor.

**Comment by UbaserB** [comment]
The microphone sprite looks weird. The top looks like a grater and there's already a microphone sprite in the game so maybe you can tweak that one.

**Comment by SoulFN** [comment]
I think microphone should be in the reporter's backpack at the roundstart.

**Comment by noctyrnal** [comment]
> Chatfactor warning, I'll need to eyeball this code (similar to how I had to fuck with handheld radios) during the content refactor.

I can use the original microphone sprite, though my issue was that people would confuse it with the microphone instrument.

**Comment by noctyrnal** [comment]
> I think microphone should be in the reporter's backpack at the roundstart.

I'll just put it in their pockets.

**Comment by UbaserB** [comment]
> > Chatfactor warning, I'll need to eyeball this code (similar to how I had to fuck with handheld radios) during the content refactor.
> 
> I can use the original microphone sprite, though my issue was that people would confuse it with the microphone instrument.

which is why i said you should tweak the original microphone sprite

**Comment by EmoGarbage404** [comment]
just combine them

**Comment by 2013HORSEMEATSCANDAL** [comment]
if you can you should see if there's a way to make music go through these mics as well, so radio hosts can play music.

File: Resources/Prototypes/Entities/Objects/Devices/encryption_keys.yml
Comment (Partmedia) [inline Resources/Prototypes/Entities/Objects/Devices/encryption_keys.yml:1]: Please revert this no-op change to this file.

File: Resources/Textures/Objects/Devices/communication.rsi/meta.json [hunk 1]
```diff
@@ -69,16 +69,27 @@
       "name": "off-walkietalkie-inhand-left",
       "directions": 4
     },
-	{
+	  {
```
Comment (Partmedia) [inline Resources/Textures/Objects/Devices/communication.rsi/meta.json:72]: Please fix the whitespace here.

**Review by Partmedia — COMMENTED** [review]
I've highlighted a few nitpicks on the changes.

The microphone sprite in the PR description looks comically large; I don't know if you had a chance to change this yet.

**Comment by Peptide90** [comment]
> if you can you should see if there's a way to make music go through these mics as well, so radio hosts can play music.

Probably out of scope for this PR but took me a second to realise you meant MIDIs not actual music.

Detecting a midi in range of the microphone then broadcasting it would be good. At a reduced volume / range and with a radio / static effect over the top to really capture that retro futurism. Probably not the place to discuss it but just thought I'd add my two cents.

**Pull request closed without merging**